tests/integration/crud/rest/test_rest.py

```diff
@@ -0,0 +1,205 @@
+import os
+import random
+import string
+from itertools import chain
+from pathlib import Path
+
+import numpy as np
+import pytest
+import requests
+
+from jina.executors.indexers import BaseIndexer
+
+from jina import Document
+from jina.flow import Flow
+
+
+random.seed(0)
+np.random.seed(0)
+
+@pytest.fixture
+def config(tmpdir):
+    os.environ['JINA_REST_DIR'] = str(tmpdir)
+    yield
+    del os.environ['JINA_REST_DIR']
+
+
+def send_rest_request(flow_file, endpoint, method, data):
+    json = {
+        'data': data
+    }
+    with Flow.load_config(flow_file) as flow:
+        url = f'http://0.0.0.0:{flow.port_expose}/{endpoint}'
+        r = getattr(requests, method)(url, json=json)
+
+        if r.status_code != 200:
+            # TODO status_code should be 201 for index
+            raise Exception(f'api request failed, url: {url}, status: {r.status_code}, content: {r.content} data: {data}')
+
+    return r
+
+
+def send_rest_index_request(flow_file, documents):
+    data = [document.dict() for document in documents]
+    return send_rest_request(flow_file, 'index', 'post', data)
+
+
+def send_rest_update_request(flow_file, documents):
+    data = [document.dict() for document in documents]
+    return send_rest_request(flow_file, 'update', 'put', data)
+
+
+def send_rest_delete_request(flow_file, ids):
+    return send_rest_request(flow_file, 'delete', 'delete', ids)
+
+
+def send_rest_search_request(flow_file, documents):
+    data = [document.dict() for document in documents]
+    return send_rest_request(flow_file, 'search', 'post', data)
+
+
+def random_docs(start, end):
+    documents = []
+    for j in range(start, end):
+        d = Document()
+        d.id = j
+        d.tags['id'] = j
+        d.text = ''.join(random.choice(string.ascii_lowercase) for _ in range(10)).encode('utf8')
+        d.embedding = np.random.random([10 + np.random.randint(0, 1)])
+        documents.append(d)
+    return documents
+
+
+def get_ids_to_delete(start, end):
+    return [str(idx) for idx in range(start, end)]
+
+
+
+def validate_index_size(num_indexed_docs, compound=False):
+    from jina.executors.compound import CompoundExecutor
+
+    if compound:
+        path = Path(CompoundExecutor.get_component_workspace_from_compound_workspace(os.environ['JINA_REST_DIR'],
+                                                                                     'chunk_indexer', 0))
+    else:
+        path = Path(os.environ['JINA_REST_DIR'])
+    bin_files = list(path.glob('*.bin'))
+    assert len(bin_files) > 0
+    for index_file in bin_files:
+        index = BaseIndexer.load(str(index_file))
+        assert index.size == num_indexed_docs
+
+
+@pytest.mark.parametrize('flow_file, compound', [
```

Comment:
`compound` variable is not necessary. It is coupled with the choice of flow configuration file.
```suggestion
@pytest.mark.parametrize('flow_file, compound', [
```

Reply:
Thanks, good idea! I removed it and always look in both directories when doing the validation.